### ISSUE TITLE
When defining an abstract class with generics, compilation fails.

### ISSUE THREAD

**Issue by ErikHellman** [rharter/auto-value-parcel#31]
Given the class `@AutoValue abstract class Foo<T>`, compilation fails when I include auto-value-parcel (`apt 'com.ryanharter.auto.value:auto-value-parcel:0.2-SNAPSHOT'`).

**Comment by JakeWharton** [comment]
In order to do this we need to validate the generic bounds. Your example would not be allowed unless you added something like `T extends Parcelable`.

And it's even more complicated than that since we actually need to resolve the bounds in the case where you can take a bound that isn't valid and make it valid by specifying a more concrete type:
```java
interface TableA {}
interface ViewB<V extends TableA> {
  V tableA();
}
@AutoValue class RealTableA implements TableA, Parcelable {}
@AutoValue class RealViewB implements ViewB<RealTableA>, Parcelable {}
```

**Comment by muddin1** [comment]
I get an error, `cannot find symbol class T` when I use generics in a Parcelable class.

This generic T is referenced in the extension generated class but unlike auto-value-gson and auto-value's generated implementations, this one doesn't specify the generic when extending the generic class or its auto-value implementations.

version: 0.2.4-rc2

**Comment by maxchursin** [comment]
I figured it's a difficult to fix, spending second day digging around code, I'd love to make a fix and rollout pr for it, but can somebody point me to the right direction? 
My test case is the same as mentioned in issue #48 
`@AutoValue public abstract class Stuff<T extends Parcelable> implements Parcelable {
  public abstract List<T> data();
}`
seems like the issue will be fixed if only generated class include genetic type declaration as:
`final class AutoValue_Stuff<T extends Parcelable> extends $AutoValue_Stuff {...}`
But I can't figure at which point in code it gets stripped out. 
To the point @JakeWharton brings - can't we just fail if bounds are not valid?

**Comment by rharter** [comment]
I've been meaning to get to this, just haven't yet.  You can take a look at
this PR on auto-value-gson for a good starting point.
https://github.com/rharter/auto-value-gson/pull/61

**Comment by JakeWharton** [comment]
See also: https://github.com/rharter/auto-value-parcel/compare/master...JakeWharton:jw/generical

Feel free to take this branch over.

**Comment by rharter** [comment]
This is fixed in #90
